source/extensions/filters/udp/dns_filter/dns_parser.h

```diff
@@ -0,0 +1,189 @@
+#pragma once
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/platform.h"
+#include "envoy/network/address.h"
+#include "envoy/network/listener.h"
+
+#include "common/buffer/buffer_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace DnsFilter {
+
+constexpr uint16_t DNS_RECORD_CLASS_IN = 1;
+constexpr uint16_t DNS_RECORD_TYPE_A = 1;
+constexpr uint16_t DNS_RECORD_TYPE_AAAA = 28;
+/**
+ * BaseDnsRecord contains the fields and functions common to both query and answer records.
+ */
+class BaseDnsRecord {
+public:
+  BaseDnsRecord(const std::string& rec_name, const uint16_t rec_type, const uint16_t rec_class)
+      : name_(rec_name), type_(rec_type), class_(rec_class){};
+
+  virtual ~BaseDnsRecord() = default;
+  void serializeName(Buffer::OwnedImpl& output);
+  virtual void serialize(Buffer::OwnedImpl& output) PURE;
+
+  const std::string name_;
+  const uint16_t type_;
+  const uint16_t class_;
+};
+
+/**
+ * DnsQueryRecord represents a query record parsed from a DNS request from a client. Each record
+ * contains the ID, domain requested and the flags dictating the type of record that is sought.
+ */
+class DnsQueryRecord : public BaseDnsRecord {
+
+public:
+  DnsQueryRecord(const std::string& rec_name, const uint16_t rec_type, const uint16_t rec_class)
+      : BaseDnsRecord(rec_name, rec_type, rec_class) {}
+
+  ~DnsQueryRecord() override = default;
```

Comment:
not needed as overrides virtual class, same below

Reply:
ping

Reply:
Fixed.

Reply:
Delete the entire line. It's redundant and not needed. Same elsewhere. That is what I meant.

/wait

Reply:
Done